### ISSUE TITLE
Add a setting to remove the unclear messages?

### ISSUE THREAD

**Issue by yrmvgh** [Al-Caveman/commorkers#1]
Commorkers seems to generate some messages that, to me, are unclear and extraneous. See image:

![3 lines from commorker](http://storage3.static.itmages.com/i/16/0922/h_1474505063_9034249_66f44c70a4.png)

As best as I can guess, this third message tells me that "someone" has joined the channel, but I need to go check that other channel to see who it is?

I would sooner just not see these messages. Could you add a config setting for that? I can use weechat's /filter for now. :smile_cat:

**Comment by yrmvgh** [comment]
Oh, it's meant to show up right after a /join. But because I filter those, I don't know who it is.

**Comment by Al-Caveman** [comment]
Yeah you're right, it's meant to appear after /join messages.

**Comment by Al-Caveman** [comment]
Btw I am not sure I understand what's going on. Are you suggesting a feature, or considering that this behaviour is a bug? If so, may I know what's the alternative behaviour that you desire?

**Comment by yrmvgh** [comment]
In my first post, I wasn't sure whether it was a bug or not. So yes, it is a feature request, for options that can enable / disable different kinds of messages piecemeal. EG, allow people to choose whether they want "extended lurkage", whether they want "a lurker from", and whether they want "also lurks/lurked in".

Personally, with all commorker messages enabled, and myself being a member of over 30 channels, the script can be a little spammy - but I do find it useful and interesting.

**Comment by Al-Caveman** [comment]
Note: for now, as a temporary remedy, have you considered suppressing all triggered commorkers notifications, and using /commorkers manually? E.g. if you set **plugins.var.perl.commorkers.ignore_all_triggered_notices = "on"** then you will not get any notifications at all. This way you can look around for lurkers manually by executing the command **/commorkers**, or **/commorkers NICKNAME** every time you feel the need to look around.

Back to the main point: I agree with. Having finer control to disable/enable specific classes of notifications is good. This way, when you disable /join, you can also disable "a lurker from.." along with it.

Another feature could be that, when /join is disabled, the message "a lurker from.." gets transformed into "NICKNAME is a lurker from...". Or, how about just replace "a lurker from.." by "NICKNAME is a lurker from.." for all cases?

**Comment by yrmvgh** [comment]
"NICKNAME is a lurker from.." for all cases is probably superior, actually. The only downside I can think of is when someone has a very narrow chat area due to screen size + bars, and long nicknames cause "NICKNAME is a lurker from.." to overflow into the next line... but if your chat area is that narrow you're probably used to a lot of overflow.

**Comment by Al-Caveman** [comment]
Yeah I agree. Even join messages themselves are longer :laughing:. Plus, what follows is a list of channels that could be (say) 30 channel names. So my decision on saving space here makes no sense. I should really put the NICKNAME there to avoid this needless confusion.
